### PR TITLE
use django_structlog for request logging

### PR DESCRIPTION
**Pull request by danchr**

### CODE REVIEW

File: src/os2datascanner/utils/gelf.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            raise ValueError("record too large!")
+
+        count = 0
+        message_id = hash(
+            str(datetime.datetime.now().microsecond) + socket.gethostname()
+        )
+
+        for i, offset in enumerate(range(0, len(data), chunk_size)):
+            header = struct.pack(
+                "!ccqBB", b"\x1e", b"\x0f", message_id, count, total_chunks
```
Comment (dals83) [inline src/os2datascanner/utils/gelf.py]: What is going on here?

Comment (danchr) [reply src/os2datascanner/utils/gelf.py]: This adds support for messages >8kB by using [GELF UDP chunking](https://docs.graylog.org/en/3.0/pages/gelf.html#chunking).

Comment (danchr) [reply src/os2datascanner/utils/gelf.py]: > This adds support for messages >8kB by using [GELF UDP chunking](https://docs.graylog.org/en/3.0/pages/gelf.html#chunking).

I added some comments to this effect.

**Comment by magenta-af** [comment]
I'm installing a new Ubuntu VM to try this out, but after running `install.sh` (and `setup.py develop`, just to be on the safe side), I still seem to be missing some dependencies:

```
(python-env) vagrant@ubuntu-bionic:/vagrant/src$ python3 -m os2datascanner.sites.admin.manage
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/vagrant/src/os2datascanner/sites/admin/manage.py", line 13, in <module>
    execute_from_command_line(sys.argv)
  File "/vagrant/python-env/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/vagrant/python-env/lib/python3.6/site-packages/django/core/management/__init__.py", line 308, in execute
    settings.INSTALLED_APPS
  File "/vagrant/python-env/lib/python3.6/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/vagrant/python-env/lib/python3.6/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/vagrant/python-env/lib/python3.6/site-packages/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/vagrant/src/os2datascanner/sites/admin/settings.py", line 244, in <module>
    "processor": structlog.dev.ConsoleRenderer(),
  File "/vagrant/python-env/lib/python3.6/site-packages/structlog/dev.py", line 135, in __init__
    package="colorama",
SystemError: ConsoleRenderer with `colors=True` requires the colorama package installed.  
```

File: contrib/graylog/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    image: graylog/graylog:3.0
+    volumes:
+      - graylog_journal:/usr/share/graylog/data/journal
+      - ./graylog/plugin/graylog-plugin-auth-sso-2.3.0.jar:/usr/share/graylog/plugin/graylog-plugin-auth-sso-2.3.0.jar
+    environment:
+      # CHANGE ME (must be at least 16 characters)!
+      - GRAYLOG_PASSWORD_SECRET=somepasswordpepper
+      # Password: admin
+      - GRAYLOG_ROOT_PASSWORD_SHA2=8c6976e5b5410415bde908bd4dee15dfb167a9c873fc4bb8a81f6f2ab448a918
+      - GRAYLOG_HTTP_EXTERNAL_URI=http://dionic.shared:9000/
```
Comment (magenta-af) [inline contrib/graylog/docker-compose.yml]: Can this URL be made more generic somehow? I had to change this to `http://localhost:8002` (which is forwarded to port 9000 in my Vagrant machine, which is forwarded onto who knows what in the Docker container) before any of the Graylog interface would load, but I don't understand why this is necessary.

Comment (danchr) [reply contrib/graylog/docker-compose.yml]: > Can this URL be made more generic somehow? I had to change this to `http://localhost:8002` (which is forwarded to port 9000 in my Vagrant machine, which is forwarded onto who knows what in the Docker container) before any of the Graylog interface would load, but I don't understand why this is necessary.

Right; it should be `localhost:9000` — I had to change it on my Mac due to Graylog being horrible 🤷‍♂

**Comment by danchr** [comment]
> I'm installing a new Ubuntu VM to try this out, but after running `install.sh` (and `setup.py develop`, just to be on the safe side), I still seem to be missing some dependencies:
> …

Apparently, `structlog.dev.ConsoleRenderer` doesn't actually work with the listed dependencies, unless you explicitly disable color. I posted a PR — hynek/structlog#215 — to fix that.

**Pull request merged**